packages/flutter/lib/src/rendering/sliver_padding.dart

```diff
@@ -92,38 +92,34 @@ class RenderSliverPadding extends RenderSliver with RenderObjectWithChildMixin<R
     return null;
   }
 
-  /// The padding in the cross-axis direction on the side to the right when
-  /// facing away from the zero scroll offset in the scroll axis direction. (In
-  /// other words, for a vertical downwards-growing list, the padding on the
+  /// The padding in the cross-axis direction on the side closest to the origin.
+  /// (In other words, for a vertical downwards-growing list, the padding on the
   /// left.)
   ///
   /// Only valid after layout has started, since before layout the render object
   /// doesn't know what direction it will be laid out in.
-  double get startPadding {
+  double get leadingPadding {
     assert(constraints != null);
     assert(constraints.axisDirection != null);
     assert(constraints.growthDirection != null);
     switch (applyGrowthDirectionToAxisDirection(constraints.axisDirection, constraints.growthDirection)) {
       case AxisDirection.up:
-        return padding.right;
-      case AxisDirection.right:
-        return padding.top;
       case AxisDirection.down:
         return padding.left;
       case AxisDirection.left:
-        return padding.bottom;
+      case AxisDirection.right:
+        return padding.top;
     }
     return null;
   }
```

Comment:
I agree that left should mean left. If we're going up, then the "start" is on the right, no? And "end" on the left. Maybe the bug is with the code that uses these functions?